agent/session_persister.py

```
@@ -0,0 +1,479 @@
+            func = tool["function"]
+            formatted_tool = {
+                "name": func["name"],
+                "description": func.get("description", ""),
+                "parameters": func.get("parameters", {}),
+                "required": None,  # Match the format in the example
+            }
+            formatted_tools.append(formatted_tool)
+        return json.dumps(formatted_tools, ensure_ascii=False)
+
+    def _convert_to_trajectory_format(
+        self, messages: List[Dict[str, Any]], user_query: str, completed: bool
```

Ruff (0.15.2)

[error] Boolean-typed positional argument in function definition

(FBT001)